From e517ef81917d40dba7bcc13f42c35815e38753a9 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 11 Dec 2024 10:31:17 +0800
Subject: [PATCH 243/301] AOSCOS: drm: radeon: use rdev_to_drm(rdev)

Per commit fb1b5e1dd53f ("drm/radeon: change rdev->ddev to
rdev_to_drm(rdev)"), members of the struct `rdev' is now accessed with
the `rdev_to_drm()' helper function.

Follow this commit and revise the changes made in "AOSCOS: drm/radeon:
recover the GPU if it fails at resume"

Fixes: "AOSCOS: drm/radeon: recover the GPU if it fails at resume"
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/gpu/drm/radeon/radeon_device.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/radeon/radeon_device.c b/drivers/gpu/drm/radeon/radeon_device.c
index 10d2ae804de2..7c80605afecd 100644
--- a/drivers/gpu/drm/radeon/radeon_device.c
+++ b/drivers/gpu/drm/radeon/radeon_device.c
@@ -1659,7 +1659,7 @@ void radeon_recover_callback(struct work_struct *work)
 	radeon_restore_bios_scratch_regs(rdev);
 	radeon_ib_ring_tests(rdev);
 	radeon_pm_resume(rdev);
-	drm_helper_resume_force_mode(rdev->ddev);
+	drm_helper_resume_force_mode(rdev_to_drm(rdev));
 	ttm_bo_unlock_delayed_workqueue(&rdev->mman.bdev, resched);
 	up_write(&rdev->exclusive_lock);
 }
-- 
2.48.1

